WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentViewHolder.kt

```diff
@@ -34,16 +36,27 @@ sealed class NewSiteCreationSegmentViewHolder(internal val parent: ViewGroup, @L
             uiState as SegmentUiState
             title.text = uiState.title
             subtitle.text = uiState.subtitle
-            imageManager.load(icon, IMAGE, uiState.iconUrl)
-            // TODO: Check with designers on how to handle icon url or color missing, this is only temporary
-            // TODO: https://github.com/wordpress-mobile/WordPress-Android/issues/8864
-            val blackColor = "#000000"
-            val iconColor = if (uiState.iconColor.isNotBlank()) uiState.iconColor else blackColor
-            try {
-                icon.setColorFilter(Color.parseColor(iconColor), PorterDuff.Mode.SRC_IN)
-            } catch (e: IllegalArgumentException) {
-                AppLog.e(AppLog.T.SITE_CREATION, "Error parsing segment icon color: ${uiState.iconColor}")
-            }
+            imageManager.loadWithResultListener(
+                    icon,
+                    ICON,
+                    uiState.iconUrl,
+                    null,
+                    object : RequestListener<Drawable> {
```

Comment:
This is a really good question as well, but tbh I'm not 100% sure. From my understanding the requestListener is tied to the request, which is automatically canceled whenever clear()/into(View) is called or whenever the target view is detached (see the clearOnDetach() method is called on each request in ImageManager). It's difficult to track down - is there any specific reason why you think Glide might keep a reference to the listener?

Reply:
I'll update the ImageManager to keep just a WeakReference, so we can be sure.

Reply:
Hmm actually, it's not that simple. I thought I'd just create a weakReference in ImageManager.attachRequestListener(..). However, if I'm not missing something, the listener wouldn't have any strong reference and would be automatically collected (https://github.com/wordpress-mobile/WordPress-Android/compare/issue/9105-segments-add-placeholder...issue/9105-weak-reference?expand=1)... We could use a SoftReference I guess, but it doesn't feel right. Wdyt?

Reply:
>Hmm.. I guess we should trust that the library is doing the right thing since they don't have documentation telling us to do anything about it. Maybe we should just keep this in mind, but leave as is until we experience any issues with it.

I agree.

>I can merge this PR as is if you're OK with it @malinajirka, I just wanted to check in with you about these 2 before that.

👍 Thanks!